api/app/fireweather_bot/hourly_actuals.py

```diff
@@ -21,100 +23,116 @@
+        precipitation=hourly_reading.precipitation,
+        dewpoint=hourly_reading.dewpoint,
+        ffmc=hourly_reading.ffmc,
+        isi=hourly_reading.isi,
+        fwi=hourly_reading.fwi,
+    )
+
+
+def validate_metric(value, low, high):
+    """ Validate metric with it's range of accepted values """
+    return low <= value <= high
+
+
+def get_hourly_readings_synchronously_wfwx(station_codes, start_date, end_date):
+    """ Synchronously get hourly readings for given station codes and date range from wfwx """
+
+    async def authenticate_and_get_hourly_readings():
+        async with ClientSession() as session:
+            header = await wildfire_one.get_auth_header(session)
+            return wildfire_one.get_hourly_readings(session, header,
+                                                    station_codes, start_date, end_date)
+
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    return loop.run_until_complete(authenticate_and_get_hourly_readings())
+
+
+class HourlyActualsBot():
     """ Bot that downloads the hourly actuals from the wildfire website and stores it in a database. """
 
     def __init__(self):
         self.now = app.time_utils.get_pst_now()
 
-    def _get_start_date(self) -> int:
+    def _get_start_date(self) -> datetime:
         """ Return time N hour ago. E.g. if it's 17h15 now, we'd get YYYYMMDD16. The intention is that
         this bot runs every hour, so if we ask for everything from an hour back, we should be fine.
         However, just to be on the safe side, we're asking for the last three hours - just in case there
         was a station that came in late, or if for whatever reason we missed a run.
         """
         hour_ago = self.now - timedelta(hours=3)
-        return int(hour_ago.strftime('%Y%m%d%H'))
+        return hour_ago
 
-    def _get_end_date(self) -> int:
+    def _get_end_date(self) -> datetime:
         """ Return now. E.g. if it's 17h15 now, we'd get YYYYMMDD17 """
-        return int(self.now.strftime('%Y%m%d%H'))
-
-    def process_csv(self, filename: str):
-        with open(filename, 'r') as csv_file:
-            data_df = pd.read_csv(csv_file)
-        station_codes = get_station_names_to_codes()
-        # drop any rows where 'display_name' is not found in the station_codes lookup:
-        data_df.drop(index=data_df[~data_df['display_name'].isin(
-            station_codes.keys())].index, inplace=True)
-        # replace 'display_name' column (station name) in df with station_id
-        # and rename the column appropriately
-        data_df['display_name'].replace(station_codes, inplace=True)
-        data_df.rename(columns={'display_name': 'station_code'}, inplace=True)
-
-        # write to database using _session's engine
+        return self.now
+
+    async def run_wfwx(self):
+        """ Entry point for running the bot """
+        async with ClientSession(connector=TCPConnector(limit=100)) as session:
```

Comment:
We shouldn't need a limit this high? There should only be one connection at a time, unless we're doing parallel calls?

Reply:
We're using this session across all fetch hourly calls

Reply:
Changed it to a limit of 10 which seems to work.